### PR TITLE
Frontend feat state

### LINKED ISSUE #70: Redux, RxJS Setup
Redux, RxJS Setup

### PR DESCRIPTION
**Pull request by obi-wil**
**Client dependencies added**

Basic store configuration and search bar slice awaiting for plugin with the components.

Some boilerplate will come very soon to the store, please review promptly.

Thanks!

Close #70

### CODE REVIEW

**Review by Dalton-Klein — COMMENTED** [review]
Hey Wilfredo, looks like most of this are state files, but there are some conflicts with components. It doesn't look like conflicts vary too much, but we need a working version of the app to present to Mo today.

**Comment by obi-wil** [comment]
Conflicts solved. Ready to merge.

File: client/trademon/src/components/containerComponents/searchResultsContainer.tsx [hunk 1]
```diff
@@ -1,39 +1,35 @@
...
+import React, { useEffect, useState } from 'react';
+import { getTrades } from '../../utils/rest';
+import StandardTile from '../tileComponents/standardTileComponent';
 interface trades {
-  tradeID:number,
-  numViews:number, 
-  seller:string, 
-  pokeNum:number, 
-  pokeName:string, 
-  pokeGen:number,
```
Comment (gsjuradog) [inline client/trademon/src/components/containerComponents/searchResultsContainer.tsx:12]: this changes are just form rigth? is that the reason of the conflict?

**Review by Dalton-Klein — APPROVED** [review]
Lets merge this now!

**Pull request merged**